src/init/map.init.js

```diff
@@ -26,77 +24,45 @@ export const initMap = async (
 
     const spawnpointsCharacters = {};
 
-    for (const layer of layers) {
-        const isStaticObject = objectConfig.static.includes(layer.name);
-        const isInteractionObject = objectConfig.interactionObjects.includes(
-            layer.name
-        );
+    layers.forEach(layer => {
+        const { name, objects, properties } = layer;
+        const isStaticObject = objectConfig.static.includes(name);
+        const isInteractionObject = objectConfig.interactionObjects.includes(name);
 
         if (isStaticObject || isInteractionObject) {
-            const isStaticProp = layer?.properties?.filter(
-                (prop) => prop.name === 'isStatic'
-            );
-            const isStatic =
-                isStaticProp.length > 0 ? isStaticProp[0].value : true;
+            const isStatic = properties?.some(prop => prop.name === 'isStatic' && prop.value) ?? true;
 
-            for (const staticBodyObj of layer.objects) {
```

Comment:
making this change might be bad for performance because using array.forEach will always run through the entire array. I used for of here to early return which is not possible by using forEach